lib/matplotlib/tests/test_mlab.py

```diff
@@ -2103,6 +2104,14 @@ def test_specgram_phase(self):
         assert spec.shape[0] == freqs.shape[0]
         assert spec.shape[1] == self.t_specgram.shape[0]
 
+    def test_specgram_warn_only1seg(self):
+        """Warning should be raised if len(x) <= NFFT. """
+        with warnings.catch_warnings(record=True) as w:
+            mlab.specgram(x=self.y, NFFT=len(self.y), Fs=self.Fs)
```

Comment:
Just before this line will need a line with

`warnings.simplefilter('always')`